src/test/java/user/TC_UserDelete.java

```diff
@@ -0,0 +1,129 @@
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+import org.testng.internal.collections.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TC_UserDelete {
+
+	private final static String NAME = "Anton";
+
+	private SoftAssert softAssert;
+	private UserDto userDto;
+
+	private List<UserDto> userDtoList;
+
+	@BeforeClass
+	public void initUserListForRemoving() {
+		userDtoList = new ArrayList<>();
+	}
+
+	@BeforeMethod
+	public void initSoftAssert() {
+		softAssert = new SoftAssert();
+		userDto = new UserDto();
+	}
+
+	@Test
+	public void verifyRemoveUserTest() {
+		String zipCode = RandomUtils.nextInt(10000, 99999) + "4";
+		ZipCodeService.addZipCodes(Lists.newArrayList(zipCode));
+
+		String uniqueName = NAME + RandomStringUtils.randomAlphabetic(4);
+		userDto.setName(uniqueName);
+		userDto.setAge(RandomUtils.nextInt(1, 99));
+		userDto.setSex(Sex.values()[RandomUtils.nextInt(0, Sex.values().length)]);
+		userDto.setZipCode(zipCode);
+
+		UserService.createUser(userDto);
+
+		int statusCode = UserService.deleteUser(userDto);
+		softAssert.assertEquals(statusCode, HttpStatus.SC_NO_CONTENT,
+				"Response code is NOT 204 when removing a user.");
+
+		List<UserDto> users = UserService.getUsers().second();
+		softAssert.assertFalse(users.contains(userDto), "User is NOT deleted.");
+
+		Pair<Integer, List<String>> availableZipCodes = ZipCodeService.getAvailableZipCodes();
+		softAssert.assertTrue(availableZipCodes.second().contains(zipCode),
+				"Zip code is NOT returned in list of available zip codes when removing a user.");
+
+		softAssert.assertAll();
+	}
+
+	@Test
+	public void verifyRemoveUserWithRequiredFieldsTest() {
+		String zipCode = RandomUtils.nextInt(10000, 99999) + "4";
+		ZipCodeService.addZipCodes(Lists.newArrayList(zipCode));
+
+		String uniqueName = NAME + RandomStringUtils.randomAlphabetic(4);
+		userDto.setName(uniqueName);
+		userDto.setAge(RandomUtils.nextInt(1, 99));
+		userDto.setSex(Sex.values()[RandomUtils.nextInt(0, Sex.values().length)]);
+		userDto.setZipCode(zipCode);
+
+		UserService.createUser(userDto);
+
+		UserDto deleteUser = new UserDto();
+		deleteUser.setName(userDto.getName());
+		deleteUser.setSex(userDto.getSex());
+
+		int statusCode = UserService.deleteUser(deleteUser);
+		softAssert.assertEquals(statusCode, HttpStatus.SC_NO_CONTENT,
+				"Response code is NOT 204 when removing a user with required fields.");
+
+		softAssert.assertEquals(UserService.findUsersByName(uniqueName).size(), 0,
+				"User with required fields is NOT deleted.");
+
+		Pair<Integer, List<String>> availableZipCodes = ZipCodeService.getAvailableZipCodes();
+		softAssert.assertTrue(availableZipCodes.second().contains(zipCode),
+				"Zip code is NOT returned in list of available zip codes when removing a user with required fields.");
+
+		softAssert.assertAll();
+	}
+
+	@Test
+	public void verifyRemoveUserAnyRequiredFieldMissedTest() {
```

Comment:
Let's add assertion that zip code is not returned in available list

Reply:
Added one more assertion.